SDWebImage/UIView+WebCache.m

```diff
@@ -110,30 +111,38 @@ - (void)sd_internalSetImageWithURL:(nullable NSURL *)url
             manager = [SDWebImageManager sharedManager];
         }
         
-        __weak __typeof(self)wself = self;
+        __weak __typeof(self) wself = self;
         SDWebImageDownloaderProgressBlock combinedProgressBlock = ^(NSInteger receivedSize, NSInteger expectedSize, NSURL * _Nullable targetURL) {
-            wself.sd_imageProgress.totalUnitCount = expectedSize;
-            wself.sd_imageProgress.completedUnitCount = receivedSize;
+            __strong __typeof(wself) sself = wself;
+            if (!sself) { return ; }
+            if ([url isEqual:sself.sd_imageURL]) {
+                sself.sd_imageProgress.totalUnitCount = expectedSize;
+                sself.sd_imageProgress.completedUnitCount = receivedSize;
+            }
```

Comment:
this is odd, we are masking the setters behind a url check, but we always call the `progressBlock`. is this right?